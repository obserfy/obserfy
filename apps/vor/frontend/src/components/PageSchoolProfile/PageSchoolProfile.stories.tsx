import { Meta, Story } from "@storybook/react"
import PageSchoolProfile, { PageSchoolProfileProps } from "./PageSchoolProfile"

export default {
  title: "Core/PageSchoolProfile",
  component: PageSchoolProfile,
  parameters: {
    componentSubtitle: "Just a simple PageSchoolProfile",
  },
} as Meta

const Template: Story<PageSchoolProfileProps> = (args) => (
  <PageSchoolProfile {...args} />
)

export const Default = Template.bind({})
Default.args = {}
