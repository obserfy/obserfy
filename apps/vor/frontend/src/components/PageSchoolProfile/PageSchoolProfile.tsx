import { t } from "@lingui/macro"
import { FC, useState } from "react"
import { Box, Card } from "theme-ui"
import { useGetSchool } from "../../hooks/api/schools/useGetSchool"
import usePatchSchool from "../../hooks/api/schools/usePatchSchool"
import { getSchoolId } from "../../hooks/schoolIdState"
import useVisibilityState from "../../hooks/useVisibilityState"
import { ADMIN_URL } from "../../routes"
import DataBox from "../DataBox/DataBox"
import Dialog from "../Dialog/Dialog"
import DialogHeader from "../DialogHeader/DialogHeader"
import Input from "../Input/Input"
import TopBar, { breadCrumb } from "../TopBar/TopBar"
import Typography from "../Typography/Typography"

export interface PageSchoolProfileProps {}
const PageSchoolProfile: FC<PageSchoolProfileProps> = () => {
  const school = useGetSchool()

  return (
    <Box mx="auto" sx={{ maxWidth: "maxWidth.sm" }}>
      <TopBar
        breadcrumbs={[
          breadCrumb("Admin", ADMIN_URL),
          breadCrumb("School Profile"),
        ]}
      />

      <Box m={3}>
        <Typography.H5>School Profile</Typography.H5>
      </Box>

      <Card variant="responsive">
        <NameDataBox originalValue={school.data?.name} />
      </Card>
    </Box>
  )
}

const NameDataBox: FC<{ originalValue?: string }> = ({
  originalValue = "",
}) => {
  const editDialog = useVisibilityState()

  return (
    <>
      <DataBox
        label={t`School name`}
        value={originalValue}
        onEditClick={editDialog.show}
      />

      {editDialog.visible && (
        <EditNameDialog
          originalValue={originalValue}
          onDismiss={editDialog.hide}
        />
      )}
    </>
  )
}

const EditNameDialog: FC<{ originalValue: string; onDismiss: () => void }> = ({
  originalValue,
  onDismiss,
}) => {
  const [value, setValue] = useState(originalValue)
  const patchSchool = usePatchSchool(getSchoolId())

  const handleSave = () => {
    try {
      patchSchool.mutate({ name: value })
      onDismiss()
    } catch (e) {
      Sentry.captureException(e)
    }
  }

  return (
    <Dialog>
      <DialogHeader
        title={t`Edit School Name`}
        onCancel={onDismiss}
        onAccept={handleSave}
      />

      <Box px={3} pt={2} pb={3} sx={{ backgroundColor: "background" }}>
        <Input
          label={t`Name`}
          sx={{ width: "100%" }}
          value={value}
          onChange={(e) => setValue(e.target.value)}
        />
      </Box>
    </Dialog>
  )
}

export default PageSchoolProfile
