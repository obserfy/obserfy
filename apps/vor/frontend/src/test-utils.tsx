/* eslint-disable import/no-extraneous-dependencies */
import { FC, ReactElement } from "react"
import { QueryClient, QueryClientProvider } from "react-query"
import { ThemeProvider, Box, Theme } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import { I18nProvider } from "@lingui/react"
import { i18n } from "@lingui/core"
import { en } from "make-plural/plurals"
import theme from "./gatsby-plugin-theme-ui"
// @ts-ignore
import enCatalog from "../i18n/lingui/en/messages.js"

// any used here because the catalogs are generated b lingui
// not our responsibility
i18n.loadLocaleData("en", { plurals: en })
i18n.load("en", enCatalog.messages as any)
i18n.activate("en")

const queryClient = new QueryClient()
const history = createHistory(createMemorySource("/"))

const AllProviders: FC = ({ children }) => {
  return (
    <I18nProvider i18n={i18n}>
      <QueryClientProvider client={queryClient}>
        <LocationProvider history={history}>
          <ThemeProvider theme={theme as Theme}>
            <Box sx={{ fontSize: [16, 20] }}>{children}</Box>
          </ThemeProvider>
        </LocationProvider>
      </QueryClientProvider>
    </I18nProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
