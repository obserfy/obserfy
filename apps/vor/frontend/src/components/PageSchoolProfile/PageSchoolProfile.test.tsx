import { render } from "../../test-utils"
import PageSchoolProfile from "./PageSchoolProfile"

describe("PageSchoolProfile", () => {
  it("should render correctly", () => {
    const { container } = render(<PageSchoolProfile />)
    expect(container).toMatchSnapshot()
  })
})
