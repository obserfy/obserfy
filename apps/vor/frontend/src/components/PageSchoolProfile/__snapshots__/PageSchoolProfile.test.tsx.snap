// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageSchoolProfile should render correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    font-size: 20px;
  }
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: auto;
  margin-right: auto;
  max-width: 640px;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 48px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-3 {
  margin: 8px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary);
  border: 0;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  box-sizing: border-box;
  font-size: 14px;
  color: var(--theme-ui-colors-textPrimary);
  background-color: transparent;
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  touch-action: manipulation;
  box-shadow: none;
  border-color: var(--theme-ui-colors-border);
  border-style: solid;
  border-width: 1px;
  line-height: 1;
  border: none;
  padding: 4px;
}

.emotion-4:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-4:hover,
.emotion-4:focus {
  background-color: var(--theme-ui-colors-primaryLighter);
  border-color: var(--theme-ui-colors-primary);
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  fill: var(--theme-ui-colors-icon);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20px;
  height: 20px;
  fill: var(--theme-ui-colors-textMediumEmphasis);
}

.emotion-6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow-x: auto;
}

.emotion-7 {
  display: inline-block;
  font-size: 13px;
  color: var(--theme-ui-colors-textMediumEmphasis);
  white-space: nowrap;
}

.emotion-7:last-child {
  color: var(--theme-ui-colors-textPrimary);
}

.emotion-7:last-child>.splitter {
  display: none;
}

.emotion-8:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-9 {
  margin-left: 4px;
  margin-right: 4px;
}

.emotion-10 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: inline-block;
  font-size: 13px;
  color: var(--theme-ui-colors-textMediumEmphasis);
  white-space: nowrap;
}

.emotion-10:last-child {
  color: var(--theme-ui-colors-textPrimary);
}

.emotion-10:last-child>.splitter {
  display: none;
}

.emotion-11 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin: 16px;
}

.emotion-12 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-weight: bold;
  line-height: heading;
  color: var(--theme-ui-colors-text);
  text-rendering: optimizeLegibility;
  font-size: 24px;
}

.emotion-13 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  background-color: var(--theme-ui-colors-surface);
  border-radius: 0;
  box-shadow: rgba(0, 0, 0, 0.1) 0px 0px 1px,rgba(0, 0, 0, 0.27) 0px 2px 4px -2px;
  margin-left: 0;
  margin-right: 0;
}

@media screen and (min-width: 40em) {
  .emotion-13 {
    border-radius: 6px;
    margin-left: 16px;
    margin-right: 16px;
  }
}

.emotion-14 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 16px;
  padding-bottom: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.emotion-15 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.emotion-16 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
  margin-bottom: 4px;
  color: var(--theme-ui-colors-textMediumEmphasis);
  line-height: 1;
  font-size: 13px;
}

.emotion-17 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
}

.emotion-18 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary);
  border: 0;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  box-sizing: border-box;
  font-size: 14px;
  color: var(--theme-ui-colors-textPrimary);
  background-color: var(--theme-ui-colors-surface);
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  touch-action: manipulation;
  box-shadow: none;
  border-color: var(--theme-ui-colors-border);
  border-style: solid;
  border-width: 1px;
  line-height: 1;
  margin-left: auto;
  padding-left: 8px;
  padding-right: 8px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-18:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-18:hover,
.emotion-18:focus {
  background-color: var(--theme-ui-colors-primaryLighter);
  border-color: var(--theme-ui-colors-primary);
}

.emotion-18:hover:disabled,
.emotion-18:focus:disabled {
  background-color: transparent;
  border-color: var(--theme-ui-colors-border);
}

.emotion-19 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  fill: var(--theme-ui-colors-icon);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20px;
  height: 20px;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        <a
          class="emotion-3"
          data-cy="back-button"
          href="/dashboard/admin"
        >
          <button
            class="emotion-4"
          >
            <div
              class="emotion-5"
            />
          </button>
        </a>
        <div
          class="emotion-6"
        >
          <a
            class="emotion-7"
            href="/dashboard/admin"
          >
            <span
              class="emotion-8"
            >
              Admin
            </span>
            <span
              class="splitter emotion-9"
            >
              /
            </span>
          </a>
          <div
            class="emotion-10"
          >
            School Profile
            <span
              class="splitter"
            >
              /
            </span>
          </div>
        </div>
      </div>
      <div
        class="emotion-11"
      >
        <h5
          class="emotion-12"
        >
          School Profile
        </h5>
      </div>
      <div
        class="emotion-13"
      >
        <div
          class="emotion-14"
        >
          <div
            class="emotion-15"
          >
            <p
              class="emotion-16"
            >
              School name
            </p>
            <p
              class="emotion-17"
            />
          </div>
          <button
            aria-label="edit-school-name"
            class="emotion-18"
          >
            <div
              class="emotion-19"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
